File: src/main/java/com/wgzhao/addax/admin/repository/EtlSourceRepo.java
package com.wgzhao.addax.admin.repository;

import com.wgzhao.addax.admin.model.EtlSource;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface EtlSourceRepo
        extends JpaRepository<EtlSource, Integer> {

    Integer countByEnabled(boolean b);

    List<EtlSource> findAllByEnabled(boolean b);

    boolean existsByCode(String code);

    List<EtlSource> findByEnabled(boolean b);
}

File: src/main/kotlin/com/wgzhao/addax/admin/repository/EtlSourceRepo.kt
package com.wgzhao.addax.admin.repository

import com.wgzhao.addax.admin.model.EtlSource
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface EtlSourceRepo

    : JpaRepository<EtlSource?, Int?> {
    fun countByEnabled(b: Boolean): Int?

    fun findAllByEnabled(b: Boolean): MutableList<EtlSource?>?

    fun existsByCode(code: String?): Boolean
    fun findByEnabled(bool: Boolean): MutableList<EtlSource>?

}
